functional_tests/scylla_operator/test_functional.py

```diff
@@ -860,3 +861,11 @@ def test_cloud_bundle_connectivity_cqlsh(db_cluster: ScyllaPodCluster):
 
     assert not res.stderr
     assert '(1 rows)' in res.stdout
+
+
+def test_can_recover_from_fatal_pod_termination(db_cluster):
+    target_node = db_cluster.nodes[-1]
+    experiment = PodFailureExperiment(pod=target_node, duration="30s")
+    experiment.start()
+    experiment.wait_until_finished(60)
+    db_cluster.wait_for_nodes_up_and_normal(nodes=[target_node], verification_node=target_node)
```

Comment:
This is test and we should check more things here. For now, it doesn't make sure that anything really happened.
It must fail if everything was just idle.
For example, using scylla image with non-semver compatible docker tag any update to the Scylla spec won't make any effect and your test must catch it, but it won't.

Reply:
why this test should be about the non-semver "feature" ? 

I agree it doesn't verify the experiment was working, i.e. before it's waiting, it can check the state of the pod is failing/fetching image. 

and then wait for the experiment to be over, and test the state of the cluster/pod is back to normal.

I think it's enough for this test, keep in mind it's just a playground, before we start calling this code from Nemesis.

Reply:
I added verification that pod loses `Ready` status